### PR TITLE
More Mesheryctl Error Wrappers

### LINKED ISSUE #8204: [mesheryctl] Review mesheryctl filter commands for replacing errors with Meshkit error codes
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Desired Behavior
Review `mesheryctl filter` subcommands to replace errors with appropriate Meshkit error codes.

#### Mockups

---

#### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Philip-21**
**Notes for Reviewers**

This PR fixes #8204  and continuation of #8256 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/pkg/utils/error.go [hunk 1]
```diff
@@ -235,35 +243,25 @@ func formatError(msg string, cmd cmdType) string {
...
 func ErrFailRequest(err error) error {
 	return errors.New(ErrFailRequestCode, errors.Alert, []string{"Failed to make a request"},
-		[]string{err.Error()}, []string{"Meshery server isn't running or not reachable"}, []string{"Please check your kubernetes cluster is running or not and Run `mesheryctl system restart`"})
+		[]string{err.Error()}, []string{"Meshery server isn't running or not reachable"}, []string{"Please check if your kubernetes cluster is running or your network connection and Run `mesheryctl system restart`"})
 }
 
 func ErrUnauthenticated() error {
 	return errors.New(ErrUnauthenticatedCode, errors.Alert, []string{"UnAuthenticated User"},
-		[]string{"You are UnAuthenticated to access any resource"}, []string{"You haven't login to meshery"}, []string{"Login with `mesheryctl system login`"})
+		[]string{"You are UnAuthenticated to access any resource"}, []string{"You haven't logged in to meshery"}, []string{"Please Login with `mesheryctl system login`"})
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/error.go:253]: Always capitalize "Meshery"

Comment (leecalcote) [inline mesheryctl/pkg/utils/error.go:253]: It sounds polite, but there are two issues with "Please":

1) you didn't lowercase the following word ("Login").
2) don't use "Please", "thank you" or any colloquial or any derogatory or any complementary or any other style language than plain, direct, concise, factual, and systematic in nature.

File: mesheryctl/pkg/utils/error.go [hunk 2]
```diff
@@ -235,35 +243,25 @@ func formatError(msg string, cmd cmdType) string {
...
-	return errors.New(ErrFailReqStatusCode, errors.Alert, []string{},
-		[]string{"Response Status Code " + strconv.Itoa(statusCode) + ", possible Server Error"}, []string{}, []string{})
-}
-
 func ErrUnmarshal(err error) error {
-	return errors.New(ErrUnmarshalCode, errors.Alert, []string{"Error unmarshalling response"},
+	return errors.New(ErrUnmarshalCode, errors.Alert, []string{"Error unmarspossiblehalling response"},
 		[]string{"Error processing JSON response from server.\n" + err.Error()},
-		[]string{"JSON format from response body is not valid"}, []string{"Check if valid JSON is given to process"})
+		[]string{"JSON format from response body is not valid"}, []string{"Check if a valid JSON is given to process"})
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/error.go:264]: Revert this. Use of "a" is improper here.

File: mesheryctl/pkg/utils/error.go [hunk 3]
```diff
@@ -283,17 +281,17 @@ func ErrMarhalling(err error) error {
...
 func ErrParsingUrl(err error) error {
 	return errors.New(ErrParsingUrlCode, errors.Fatal, []string{"Error while Paring the URL"},
-		[]string{err.Error()}, []string{"The URL Provided doesn't exit or the relative path is wrong"}, []string{"Check the correctness of URL that you have inputed"})
+		[]string{err.Error()}, []string{"The URL Provided doesn't exit or the relative path is wrong"}, []string{"Please Check the correctness of URL that you have inputed"})
 }
 
 func ErrNotFound(err error) error {
 	return errors.New(ErrNotFoundCode, errors.Fatal, []string{"Not Found"}, []string{err.Error()},
-		[]string{"The item you are searching for is not present"}, []string{"Check whether that item is present"})
+		[]string{"The item you are searching for is not present"}, []string{"Please Check if that item is present"})
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/error.go:294]: Revert

File: mesheryctl/pkg/utils/error.go [hunk 4]
```diff
@@ -306,6 +304,52 @@ func ErrInvalidNameOrID(err error) error {
...
 }
 
+func ErrAttachAuthToken(err error) error {
+	return errors.New(ErrAttachAuthTokenCode, errors.Alert, []string{"Authentication token not found"},
+		[]string{"Authentication token not found", err.Error()}, []string{"User not logged in to generate token"}, []string{"Login with `mesheryctl system login` or supply a valid user token using the --token (or -t) flag."})
+}
+
+func ErrFailReqStatus(statusCode int) error {
+	return errors.New(ErrFailReqStatusCode, errors.Alert, []string{"Failed response Server Error"},
+		[]string{"Response Status Code " + strconv.Itoa(statusCode) + ", possible Server Error"}, []string{"Bad api call"}, []string{"Check your network connection and check the status of meshery server via 'mesheryctl system status'"})
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/error.go:314]: 1. Don't capitalize Server Error.
2. Don't use "bad" as an adjective here. Use "Invalid" instead.
3. Do capitalize API.
4. Always capitalize Meshery and its components' names: Meshery Server.
5. Don't use single quotes, when you should be using backticks: `mesheryctl system status`.

File: mesheryctl/pkg/utils/error.go [hunk 5]
```diff
@@ -306,6 +304,52 @@ func ErrInvalidNameOrID(err error) error {
...
+}
+
+func ErrProcessingConfig(err error) error {
+	return errors.New(ErrProcessingConfigCode, errors.Alert, []string{"Error processing config"},
+		[]string{"Error processing config", err.Error()}, []string{"Couldn't load mesh config"}, []string{"Ensure mesheryctl has the right configurations"})
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/error.go:319]: 1. Be consistent in reference to config and "mesh config" - use "meshconfig".
2. "mesh config" is one word: "meshconfig"
3. "configurations" should not be plural.
4. Don't use "right" as the descriptor, but use "valid" instead.

File: mesheryctl/pkg/utils/error.go [hunk 6]
```diff
@@ -306,6 +304,52 @@ func ErrInvalidNameOrID(err error) error {
...
+}
+
+func ErrMarshalIndent(err error) error {
+	return errors.New(ErrMarshalIndentCode, errors.Alert,
+		[]string{"Error indenting json body"}, []string{err.Error()}, []string{"Can't marshal indent body filters due to invalid content"}, []string{"Please check the data structure you are providing for Marshalling"})
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/error.go:324]: 1. Don't say "can't". Use "Unable to".
2. What does "marshal indent body filters" mean? Rephrase this.
3. Don't say "Please". 
4. Don't say "You".
5. Don't capitalize "Marshalling".
6.

File: mesheryctl/pkg/utils/error.go [hunk 7]
```diff
@@ -306,6 +304,52 @@ func ErrInvalidNameOrID(err error) error {
...
+}
+
+func ErrResponseStatusBody(statusCode int, body string) error {
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/error.go:327]: I'm not going to review the rest. There are too many grammatical mistakes here, @Philip-21. Find someone with grammatically correct English to partner with or... use ChatGPT(?).

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by Philip-21** [comment]
@leecalcote thank you I have  noted these changes

File: mesheryctl/pkg/utils/error.go (hunk 6 above)
Comment (Philip-21) [reply mesheryctl/pkg/utils/error.go:324]: Marshall Indent formats a value to a proper json format , but i will use a simpler and more concise word

**Comment by vishalvivekm** [comment]
@Philip-21 Let's discuss this on Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

File: mesheryctl/internal/cli/root/app/app.go [hunk 8]
```diff
@@ -69,20 +69,17 @@ func init() {
 func getSourceTypes() error {
 	mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
 	if err != nil {
-		utils.Log.Error(err)
-		return nil
+		return utils.ErrLoadConfig(err)
```
Comment (suhail34) [inline mesheryctl/internal/cli/root/app/app.go:72]: Revert this changes as error returned over here `config.GetMesheryCtl(viper.GetViper())` is already a meshkit error from config package and we need use meshkit logger here

File: mesheryctl/internal/cli/root/app/app.go [hunk 9]
```diff
@@ -69,20 +69,17 @@ func init() {
...
 	}
 	validTypesURL := mctlCfg.GetBaseMesheryURL() + "/api/application/types"
 	req, err := utils.NewRequest("GET", validTypesURL, nil)
 	if err != nil {
-		utils.Log.Error(err)
-		return nil
+		return utils.ErrCreatingRequest(err)
```
Comment (suhail34) [inline mesheryctl/internal/cli/root/app/app.go:77]: Same as above i have already refactored app and pattern package  please work on other packages

File: mesheryctl/internal/cli/root/app/app.go [hunk 10]
```diff
@@ -69,20 +69,17 @@ func init() {
...
 	}
 
 	resp, err := utils.MakeRequest(req)
 	if err != nil {
-		utils.Log.Error(err)
-		return nil
+		return utils.ErrRequestResponse(err)
```
Comment (suhail34) [inline mesheryctl/internal/cli/root/app/app.go:82]: Revert

File: mesheryctl/internal/cli/root/app/app.go [hunk 11]
```diff
@@ -92,12 +89,12 @@ func getSourceTypes() error {
 	body, err := io.ReadAll(resp.Body)
 	if err != nil {
 		utils.Log.Error(utils.ErrReadResponseBody(errors.Wrap(err, "couldn't read response from server. Please try again after some time")))
-		return nil
+		return utils.ErrReadResponseBody(err)
```
Comment (suhail34) [inline mesheryctl/internal/cli/root/app/app.go:92]: Revert

File: mesheryctl/internal/cli/root/app/app.go [hunk 12]
```diff
@@ -92,12 +89,12 @@ func getSourceTypes() error {
...
 	}
 	err = json.Unmarshal(body, &response)
 	if err != nil {
-		utils.Log.Error(utils.ErrUnmarshal(errors.Wrap(err, "couldn't process response received from server")))
-		return nil
+		utils.Log.Error(utils.ErrUnmarshal(err))
+		return utils.ErrUnmarshal(err)
```
Comment (suhail34) [inline mesheryctl/internal/cli/root/app/app.go:97]: Revert

File: mesheryctl/internal/cli/root/app/import.go [hunk 13]
```diff
@@ -59,22 +55,21 @@ Example: mesheryctl app import -f ./application.yml -s "Kubernetes Manifest"`
...
+			return ErrValidSource(validSourceTypes)
 		}
 
 		app, err := importApp(sourceType, file, appURL, true)
 
 		if err != nil {
-			utils.Log.Error(err)
-			return nil
+			utils.Log.Error(ErrImportApp(err))
+			return ErrImportApp(err)
```
Comment (suhail34) [inline mesheryctl/internal/cli/root/app/import.go:72]: We don't need to return the meshkit error if we are logging it via meshkit logger

**Review by suhail34 — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/app/import.go (hunk 13 above)
Comment (Philip-21) [reply mesheryctl/internal/cli/root/app/import.go:72]: @suhail34 to clarify,  so it means throughout all the packages in mesheryctl that has any var that calls a func 
`mctlCfg, err := config.GetMesheryCtl(viper.GetViper())` the error function `ErrInvalidMeshConfig` in the config already returns the error when an error occurs. 
so i would just `utils.Log.Error(err)`

File: mesheryctl/internal/cli/root/app/app.go (hunk 10 above)
Comment (Philip-21) [reply mesheryctl/internal/cli/root/app/app.go:82]: I will make changes to the returned errors, but to log them i think it best we log the functions
like this    `utils.Log.Error(utils.ErrRequestResponse(err))` instead of `utils.Log.Error(err)`

File: mesheryctl/internal/cli/root/app/import.go (hunk 13 above)
Comment (suhail34) [reply mesheryctl/internal/cli/root/app/import.go:72]: with just utils.Log.Error(err) because it is a meshkit error so we do not need to define a meshkit error again

**Comment by leecalcote** [comment]
Please resolve all comments that have been addressed.

**Review by suhail34 — CHANGES_REQUESTED** [review]
replace error codes with replace_me placeholder as ci is fixed
Also please make changes packages wise it is hard to review everything in one go you haven't consider my previous suggestion

**Comment by Philip-21** [comment]
> replace error codes with replace_me placeholder as ci is fixed Also please make changes packages wise it is hard to review everything in one go you haven't consider my previous suggestion

@suhail34 i did consider your suggestion

**Comment by Philip-21** [comment]
I want to get this done with before the weekend. Its why i decided to implement the meshkit errors here

**Comment by Philip-21** [comment]
@suhail34 i have added new errors for system  package
